Libraries/PeriphDrivers/Include/MAX78002/spi.h

```diff
@@ -1039,6 +1028,48 @@ int MXC_SPI_SlaveTransactionAsync(mxc_spi_req_t *req);
  */
 int MXC_SPI_SlaveTransactionDMA(mxc_spi_req_t *req);
 
+/**
+ * @brief   Setup an interrupt-driven, non-blocking SPI Target transaction.
+ * 
+ * @param   spi         Pointer to SPI instance's registers.
+ * @param   tx_buffer   Pointer to transmit buffer (in terms of bytes).
+ * @param   tx_fr_len   Number of frames to transmit from transmit buffer.
+ * @param   rx_buffer   Pointer to transmit buffer (in terms of bytes).
+ * @param   rx_fr_len   Number of frames to store in recieve buffer.
+ *
+ * @return  See \ref MXC_Error_Codes for the list of error return codes.
+ */
+int MXC_SPI_TargetTransaction(mxc_spi_regs_t *spi, uint8_t *tx_buffer, uint32_t tx_fr_len,
+                              uint8_t *rx_buffer, uint32_t rx_fr_len);
+
+/**
+ * @brief   Setup a blocking SPI Target transaction.
+ * 
+ * @param   spi         Pointer to SPI instance's registers.
+ * @param   tx_buffer   Pointer to transmit buffer (in terms of bytes).
+ * @param   tx_fr_len   Number of frames to transmit from transmit buffer.
+ * @param   rx_buffer   Pointer to transmit buffer (in terms of bytes).
+ * @param   rx_fr_len   Number of frames to store in recieve buffer.
+ *
+ * @return  See \ref MXC_Error_Codes for the list of error return codes.
+ */
+int MXC_SPI_TargetTransactionB(mxc_spi_regs_t *spi, uint8_t *tx_buffer, uint32_t tx_fr_len,
```

Comment:
I don't think this naming convention makes sense.

I don't see a reason to deviate from `Transaction` (blocking), `TransactionAsync` (non-blocking), and `TransactionDMA` (non-blocking).

If we want to give users a blocking/non-blocking variations of `Async` and `DMA` then that can be handled with an additional function argument.  As-is, these new functions are changing the input arguments anyways.

Reply:
Fair. The previous names were when the Transaction functions were only interrupt-driven, and before I made the Controller/Target blocking transaction functions non-interrupt-driven. I'll change the names.

Reply:
I've updated the new controller/target transaction functions to follow the v1 naming convention.